Move build-types out of cdefault

mcgordonite opened the pull request:
Defining build-types in the cdefault file is deprecated in the latest version of the schema:
https://github.com/Open-CMSIS-Pack/devtools/blob/75512df0147e04d14fc5f1eded047d1ff5b6c89d/tools/projmgr/schemas/common.schema.json#L471-L473

mcgordonite committed:
Move build-types out of cdefaults.

This is deprecated in the latest version of the schema.

Also, update schema references to point at version 1.4.0.

brondani reviewed (commented):
I would suggest to keep the common `misc` items in the `cdefault` file:
https://github.com/Open-CMSIS-Pack/devtools/blob/75512df0147e04d14fc5f1eded047d1ff5b6c89d/tools/projmgr/schemas/common.schema.json#L477
As in this example:
https://github.com/ARM-software/CMSIS-View/blob/eventstatistic/Examples/EventStatistic/EventStatistic.ac6-cdefault.yaml

mcgordonite commented:
@brondani I've moved the common misc settings into the cdefaults file. I've also updated the referenced schema versions to 1.4.0 (this version adds misc to the defaults definition). This does cause errors with the `interfaces` definitions in the clayer and cproject files, but I'm not sure exactly how to convert them to `connections`. We could either update them or remove them for now.

brondani commented:
> I'm not sure exactly how to convert them to `connections`.

The PR https://github.com/Open-CMSIS-Pack/AWS_MQTT_MutualAuth_SW_Framework/pull/4 is commenting them out, but I leave to @RobertRostohar to decide how to proceed.

RobertRostohar commented:
PR #4 has just been merged. It comments out the `interfaces` so that ctools 1.4.0 can also be used without other changes.

RobertRostohar commented:
Changes from this PR assume that ctools 1.4.0 will be used, otherwise the project might not build anymore. ctools 1.2.0 is listed in the README.md as requirement.
Are we fine by moving to 1.4.0?
It will probably break the CI (uses 1.2.0).
Which version of ctools is KSC using?

jkrech commented:
Since this repository is part of Open-CMSIS-Pack organization it was my intention that this example would always work with the latest released version of the cmsis-toolbox. 
This means we ought to update the CI tests and ReadMe accordingly.

mcgordonite commented:
> This means we ought to update the CI tests and ReadMe accordingly.

@RobertRostohar will you have time to make this update?

jkrech commented:
@mcgordonite, sorry for the lack of response.
Could you please clarify where you plan to use this example.
The intention of the repo is to keep it always running with the latest released version of the cmsis-toolbox, I am not sure whether this is a problem for your use case going forward.

mcgordonite commented:
We plan to use this and the Hello examples in the READMEs for the VS Code extensions as part of a getting started guide. This would be at least until we have a more complete MDK => csolution conversion implementation, or a good set of examples on keil.arm.com.

> The intention of the repo is to keep it always running with the latest released version of the cmsis-toolbox

One goal for the VS Code extension is to remain up to date with the latest toolbox too, so I think this will be okay. If we start to see problems, we could version the examples. E.g. use git tags to indicate which toolbox version is supported at a given commit.

JonatanAntoni commented:
I just updated to CMSIS-Toolbox 1.4.0 on main branch. 
I added the possibility to manually run the VHT workflow with a configurable version number.

mcgordonite commented:
Thanks @JonatanAntoni, this PR is no longer required.
